### PR TITLE
Add a dispatch for `is_categorical_dtype` to handle non-pandas objects

### PR DESCRIPTION
**Pull request by brandon-b-miller**
- [x] Closes https://github.com/dask/dask/issues/7054
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/dataframe/reshape.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 from .core import Series, DataFrame, map_partitions, apply_concat_apply
 from . import methods
-from .utils import is_categorical_dtype, is_scalar, has_known_categories
```
Comment (jrbourbeau) [inline dask/dataframe/reshape.py:6]: It looks like there is another function in this module using `is_categorical_dtype`. Removing this import results in `NameError: name 'is_categorical_dtype' is not defined` being raised

File: dask/dataframe/methods.py [hunk 2]
```diff
@@ -587,6 +591,10 @@ def tolist_pandas(obj):
     return obj.tolist()
 
 
+@is_categorical_dtype_dispatch.register((pd.Series, pd.Index, pd.api.extensions.ExtensionDtype, np.dtype))
+def is_categorical_dtype_pandas(obj):
+    return pd.api.types.is_categorical_dtype(obj)
+
```
Comment (jrbourbeau) [inline dask/dataframe/methods.py:597]: IIUC this change itself doesn't add support for `cudf` objects, but does allow for `cudf` to start registering it's own `is_categorical_dtype_dispatch` for things like `cudf.Series`. Is that correct, or am I missing something?

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the PR @brandon-b-miller! Just a heads up that there are a couple of code linting errors in CI. Running `black` and `flake8` locally should take care of any formatting issues (see https://docs.dask.org/en/latest/develop.html#code-formatting for more information). 

Also cc @jsignell @TomAugspurger in case either of you have thoughts on this

File: dask/dataframe/reshape.py (hunk 1 above)
Comment (brandon-b-miller) [reply dask/dataframe/reshape.py:6]: Yes - fixing this locally right now, should just be `is_categorical_dtype` -> `methods.is_categorical_dtype` I believe.

File: dask/dataframe/methods.py (hunk 2 above)
Comment (brandon-b-miller) [reply dask/dataframe/methods.py:597]: Correct, there'll be a corresponding cuDF PR that leverages this and registers the proper cuDF objects.

**Comment by kkraus14** [comment]
FYI: It looks like there's quite a bit of use of `pd.is_categorical_dtype` across the codebase that should be replaced, but we should handle that in a follow up to unblock this PR

https://github.com/dask/dask/search?p=1&q=%22is_categorical_dtype%22

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
Related cuDF PR: https://github.com/rapidsai/cudf/pull/7740

**Review by galipremsagar — APPROVED** [review]

**Review by jrbourbeau — APPROVED** [review]
Thanks @brandon-b-miller! This is in

Also, I agree with @kkraus14 that we should do some follow-up work to replace `is_categorical_dtype` more broadly throughout the codebase.

**Pull request merged**